python/ray/tests/test_autoscaler.py

```diff
@@ -105,24 +105,29 @@ def __init__(self, cache_stopped=False):
         self.ready_to_create = threading.Event()
         self.ready_to_create.set()
         self.cache_stopped = cache_stopped
+        # Many of these functions are called by node_launcher or updater in
+        # different threads. This can be treated as a global lock for everything.
+        self.lock = threading.Lock()
```

Comment:
In practice, this really reduces the amount of race conditions _especially_ when print statements/io are used.